src/OidcClient.php

```diff
@@ -218,6 +218,11 @@ public function retrieveUserInfo(OidcTokens $tokens): OidcUserData
     return new OidcUserData($data);
   }
 
+ /** {@inheritDoc} */
+  public function retrieveDataFromIdToken(OidcTokens $tokens): array {
+      return json_decode(json_encode($this->jwtHelper->getIdTokenClaims($tokens)), true);
```

Comment:
Why did you add this encode-decode step?  Is this purely to convert the object from `getIdTokenClaims` into an array? A cast (`(array)`) would be sufficient, but retrieving the property with either the property accessor or just directly with `$data->{$property}` would also be valid options.